BUILD.md

```diff
@@ -23,32 +13,71 @@ Waf should always be called from the ardupilot's root directory.
 Differently from the make-based build, with Waf there's a configure step
 to choose the board to be used (default is `sitl`):
 
+## Basic use ##
+
+There are several commands in the build system for advanced usages, but here we
+list some basic examples.
+
+### Build ArduCopter for only one board ###
+
+Here we use minlure as an example of Linux board. See more advanced usages
+below to get a list of supported boards.
+
+```bash
+./waf configure --board minlure
+./waf copter
+```
+
+This will buil all copter frames for minlure board. The configure command
+needs to be given only when you are switching from a board (or sitl) to another
+one so usually it's just one command to keep a change/test/debug cycle.
+The build will always be incremental and there's no reason to do a clean
+build
+
+Building ArduCopter Quad for Pixhawk:
+
+```bash
+./waf configure --board px4-v2
+./waf --targets bin/arducopter-quad
+```
+
+px4-v2 board supports the --upload option, so the last command could be:
+
+```bash
+./waf --targets bin/arducopter-quad --upload
+```
+
+In this case it would build and upload to a connected Pixhawk board.
+
+## Advanced use ##
+
+Waf build system is composed of commands. For example, the command below
+(`configure`) is for configuring the build.
+
```

Comment:
I don't like the new structure of the text. Cleaning up, listing and passing number of jobs should be basic usage as well. I'd say all those examples (with the ones you've added) should be part of the "Calling Waf" section. I'm okay with changing it to "Basic usage".